ale_linters/terraform/checkov.vim

```diff
@@ -0,0 +1,52 @@
+" Author: Thyme-87 <thyme-87@posteo.me>
+" Description: use checkov for providing warnings via ale
+
+call ale#Set('terraform_checkov_executable', 'checkov')
+call ale#Set('terraform_checkov_options', '-o json --quiet')
+
+function! ale_linters#terraform#checkov#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'terraform_checkov_executable')
+endfunction
+
+function! ale_linters#terraform#checkov#GetCommand(buffer) abort
+    return '%e ' . '-f %t ' . ale#Var(a:buffer, 'terraform_checkov_options')
+endfunction
+
+function! ale_linters#terraform#checkov#Handle(buffer, lines) abort
+    let l:output = []
+
+    try
```

Comment:
Something like this would be simpler:

```
let l:results = get(get(ale#util#FuzzyJSONDecode(a:lines, {}), 'results', []), 'failed_checks', [])

for l:violation in l:results
    call add(......
```

Notes:
  - Try/Catch block is unnecessary.  JSONDecode already has one inside and get() never throws.
  - The get() is called twice to retrive the 'results' and then the 'failed_checks' in sequence. In both cases an empty array is returned if the key is not present.
  - No need to check l:results for empty() since the loop latter will bail out if the array is empty.

Reply:
Okay, thanks for the explanation - is there no way, to use one get() and let it look into the json? I really don't like how it looks, to put a get() inside a get() - but I see the advantage and am okay with this.

Reply:
Unfortunately this is the best we can do with vimscript.